Clinic: Various fixes (#17911)
pingcap/docs: clinic/clinic-user-guide-for-tiup.md

        <SimpleTab groupId="clinicServer">
        <div label="Clinic Server for international users" value="clinic-us">

>         [Clinic Server for international users](https://clinic.pingcap.com): Data is stored in AWS in the US.

Reviewer: ```suggestion
        [Clinic Server for international users](https://clinic.pingcap.com): Data is stored in AWS in the United States.
```

>         [Clinic Server for international users](https://clinic.pingcap.com): Data is stored in AWS in the United States.

Reviewer: ```suggestion
        [Clinic Server for international users](https://clinic.pingcap.com): Data is stored in AWS in US regions.
```
